### ISSUE TITLE
在yolov8上使用CTKD

### ISSUE THREAD

**Issue by aowuaaa** [zhengli97/CTKD#8]
您好！阅读了你们的CTKD，对这工作十分感兴趣。我在yolov8上做蒸馏使用CTKD，开始时训练temp不会改变，发现可能是使用了amp导致的，关amp闭后temp一直增大，发现并不会经过GRL的backward。请问您有什么见解，谢谢谢谢。

**Comment by zhengli97** [comment]
这个你应该先把经过grl去backward的bug给修了吧

**Comment by aowuaaa** [comment]
问题在这，我和你们的代码一起一步一步调试对比。loss.backward()后面的运行过程一致，但到了自定义的torch.autograd.function那块并不会经过grl，我不太了解反向传播这块的代码，是有什么地方我没注意到，所以十分冒昧打扰您了。

**Comment by zhengli97** [comment]
GRL是定义在了TEMP module里面的，如果能直接用起来temp的话，就应该正常是会过GRL的

**Comment by aowuaaa** [comment]
好的好的，我自己再琢磨琢磨，大晚上还打扰了，谢谢谢谢

**Comment by zhengli97** [comment]
没事 有问题欢迎随时联系～

**Comment by ishulei** [comment]
请问楼主yolov8 + ctkd是否运行成功了呢？还有什么需要改动的地方吗
